sourcegraph/sourcegraph
# [Search Result]: Add new layout page and file preview panel

Closes https://github.com/sourcegraph/sourcegraph/issues/57800
Closes https://github.com/sourcegraph/sourcegraph/issues/57798

This PR does a few things on the search results page 
- Changes layout of the page (filters panel isn't scrollable with content anymore but scrolls independently within its own scroll block)
- Search info panel is sticky and stays on top
- But most important change is new file preview panel

| With filters | Without filters panel |
| ------- | ------- |
| <img width="1167" alt="Screenshot 2023-11-16 at 22 54 00" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/867583da-d190-4637-a7ba-39474e890dab"> | <img width="1167" alt="Screenshot 2023-11-16 at 22 54 09" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/ea228ee5-5b22-406f-97c7-50046472d879"> |

## Further steps 
- File panel preview uses inline search box to highlight search matches. This solution actually doesn't cover all cases (for example if query contains and or or operator file preview doesn't find any matches) Ideally we should use range info from the result block and highlight symbols in file preview by this data, but it requires a bit more attention and will be done in the follow PRs later. cc @fkling since you have more context on this I was hoping that we could work on this together. 
- Binary files currently fails with error since we don't support them, we shouldn't render "Preview" button in their result block.
- Search box doesn't go with caseSensitive and regex button types from the main search box. (Meaning that if your query has pattern type regex file preview search box will show up without it). There is a problem with state management of this search panel cc @fkling maybe you can help me here.

cc @taiyab all changes are behind feature flag, I checked that the current UI hasn't got any regressions from these changes. Hence I don't believe we're blocked by design here. But if you have time and have any suggestions here please share it and I can either make it here or address in follow up PRs. 

## Test plan
- Enable `newSearchNavigationUI` experimental feature flag
- Check the search result page (you should be able to see and interact with filters, aggregation UI with no regressions)
- Check file preview by clicking on preview button next to filename link in the result block
